From 3a6e2e8d1033788ef8c5263ce52f842756545a15 Mon Sep 17 00:00:00 2001
From: josekutty_kuriakose <josekutty_kuriakose@cable.comcast.com>
Date: Thu, 21 Apr 2022 02:31:34 +0000
Subject: [PATCH] Changes for keyevents
---
 RDKShell/RDKShell.cpp | 10 ++++++++++
 RDKShell/RDKShell.h   |  2 ++
 2 files changed, 12 insertions(+)
diff --git a/RDKShell/RDKShell.cpp b/RDKShell/RDKShell.cpp
index 0b387304d..29d80253e 100755
--- a/RDKShell/RDKShell.cpp
+++ b/RDKShell/RDKShell.cpp
@@ -139,6 +139,7 @@ const string WPEFramework::Plugin::RDKShell::RDKSHELL_EVENT_DEVICE_CRITICALLY_LO
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_EVENT_ON_EASTER_EGG = "onEasterEgg";
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_EVENT_ON_WILL_DESTROY = "onWillDestroy";
 const string WPEFramework::Plugin::RDKShell::RDKSHELL_EVENT_ON_SCREENSHOT_COMPLETE = "onScreenshotComplete";
+const string WPEFramework::Plugin::RDKShell::RDKSHELL_EVENT_ON_KEYEVENT = "onKeyEvent";
 
 using namespace std;
 using namespace RdkShell;
@@ -1601,6 +1602,15 @@ namespace WPEFramework {
             params["client"] = client;
             mShell.notify(RDKSHELL_EVENT_SIZE_CHANGE_COMPLETE, params);
         }
+        void RDKShell::RdkShellListener::onKeyEvent(const uint32_t keyCode, const uint32_t flags, const bool keyDown)
+        {
+            std::cout << "RDKShell onKeyEvent event received ..." << keyCode << std::endl;
+            JsonObject params;
+            params["keycode"] = keyCode;
+            params["flags"] = flags;
+            params["keyDown"] = keyDown;
+            mShell.notify(RDKSHELL_EVENT_ON_KEYEVENT, params);
+        }	
 
         // Registered methods (wrappers) begin
         uint32_t RDKShell::moveToFrontWrapper(const JsonObject& parameters, JsonObject& response)
diff --git a/RDKShell/RDKShell.h b/RDKShell/RDKShell.h
index fcbf7532c..8b2980c36 100755
--- a/RDKShell/RDKShell.h
+++ b/RDKShell/RDKShell.h
@@ -148,6 +148,7 @@ namespace WPEFramework {
             static const string RDKSHELL_EVENT_ON_EASTER_EGG;
             static const string RDKSHELL_EVENT_ON_WILL_DESTROY;
             static const string RDKSHELL_EVENT_ON_SCREENSHOT_COMPLETE;
+	    static const string RDKSHELL_EVENT_ON_KEYEVENT;
 
             void notify(const std::string& event, const JsonObject& parameters);
             void pluginEventHandler(const JsonObject& parameters);
@@ -332,6 +333,7 @@ namespace WPEFramework {
                 virtual void onEasterEgg(const std::string& name, const std::string& actionJson);
                 virtual void onPowerKey();
                 virtual void onSizeChangeComplete(const std::string& client);
+		virtual void onKeyEvent(const uint32_t keyCode, const uint32_t flags, const bool keyDown);
 
               private:
                   RDKShell& mShell;

