Date: Mon, 24 May 2021 15:42:41 -0400
From: f9d91975d9fed648c995aee8246d2b33a2641ae3 Mon Sep 17 00:00:00 2001
Subject: [PATCH] Add option to configure persistentpathpostfix
Source: COMCAST
Upstream-Status: https://github.com/rdkcentral/Thunder/pull/603
Signed-off-by: Ievgen Mutavchi <Ievgen_Mutavchi@comcast.com>


---
 cmake/project.cmake.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmake/project.cmake.in b/cmake/project.cmake.in
index 194ea5c95..e90056398 100644
--- a/cmake/project.cmake.in
+++ b/cmake/project.cmake.in
@@ -224,6 +224,11 @@ macro(write_config)
                 map_append(${plugin_config} resumed ${resumed})
             endif()
 
+            if (NOT ${persistentpathpostfix} STREQUAL "")
+                map_append(${plugin_config} persistentpathpostfix ${persistentpathpostfix})
+                unset(persistentpathpostfix)
+            endif()
+
             if (NOT ${configuration} STREQUAL "")
                 map_append(${plugin_config} configuration ${configuration})
             endif()
