From 53758afccfc690597999d270ee592a416411e0d9 Mon Sep 17 00:00:00 2001
From: Ievgen Mutavchi <Ievgen_Mutavchi@comcast.com>
Date: Wed, 7 Apr 2021 14:20:43 -0400
Subject: [PATCH] Enable BrowserConsoleLog by default

---
 Source/WPEFramework/GenericConfig.cmake | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Source/WPEFramework/GenericConfig.cmake b/Source/WPEFramework/GenericConfig.cmake
index ca5160e..29e586e 100644
--- a/Source/WPEFramework/GenericConfig.cmake
+++ b/Source/WPEFramework/GenericConfig.cmake
@@ -141,6 +141,12 @@ map()
 end()
 ans(PLUGIN_FATAL_TRACING)
 
+map()
+    kv(category "BrowserConsoleLog")
+    kv(module "Plugin_WebKitBrowser")
+    kv(enabled true)
+end()
+ans(PLUGIN_WEBKITBROWSER_TRACING)
 
 if(NOT VIRTUALINPUT)
     map()
@@ -173,6 +179,7 @@ map_append(${CONFIG} tracing ${PLUGIN_STARTUP_TRACING})
 map_append(${CONFIG} tracing ${PLUGIN_SHUTDOWN_TRACING})
 map_append(${CONFIG} tracing ${PLUGIN_NOTIFICATION_TRACING})
 map_append(${CONFIG} tracing ${PLUGIN_FATAL_TRACING})
+map_append(${CONFIG} tracing ${PLUGIN_WEBKITBROWSER_TRACING})
 endif(TRACE_SETTINGS)
 
 map_append(${PLUGIN_CONTROLLER} configuration ${PLUGIN_CONTROLLER_CONFIGURATION})
-- 
2.7.4

